src/tic-tac-toe.behavior.js

```diff
@@ -62,18 +60,20 @@ export default System.be({
 
   // Handle player moves
   makeMove: {
-    select: { self: DB.self, currentPlayer, gameStatus, event },
+    select: { self: DB.self, currentPlayer, event },
     where: [
       { Case: [DB.self, 'currentPlayer', currentPlayer] },
-      { Case: [DB.self, 'gameStatus', gameStatus] },
+      { Case: [DB.self, 'gameStatus', 'ongoing'] },
       { Case: [DB.self, `/on/click`, event] },
-      { Is: [gameStatus, 'ongoing'] },
     ],
     update: ({ self, currentPlayer, event }) => {
-      const { x, y } = event.value.target.dataset;
+      const { x, y } = event.value.target.dataset
       return [
         { Associate: [self, `${x}/${y}`, currentPlayer] },
-        { Associate: [self, 'currentPlayer', currentPlayer === 'X' ? 'O' : 'X'] }
+        { Disassociate: [self, 'currentPlayer', currentPlayer] },
+        {
+          Associate: [self, 'currentPlayer', currentPlayer === 'X' ? 'O' : 'X'],
+        },
```

Comment:
This is an only thing that required change, and caused subtle bug you've observed. What happened was that you end up with `[self, 'currentPlayer', 'X']` in first run and then with `[self, 'currentPlayer', '0']` in second run. Which meant that `renderBoard` produced two game states one per each.

Rest of the issue was than caused by this hacky code in the render loop

https://github.com/bfollington/co-dom-todomvc/blob/9fa1e167296913536d1c944fd85971aa6b12a91e/src/system.js#L226-L255

It simply queries db to get  value of `[self, `/common/ui`, ui]` and then just renders the first match ignoring the fact that there could be several matches, which would be a case because of the behavior described above. I have not verified this, but I suspect that it just happens to be that second game comes out as first match (since order is not deterministic) which is why it appears like game stops reacting after second step 😅

Reply:
We definitely could handle case where there are multiple UIs better than simply picking the first, but I have put all this in few hours so I didn't have time to think through. For what it's worth I think in terms of our OS it's not that bad it just means we have multiple views to render 😅 It will appear like forks of the gameplay